lib/kube_link_generator.rb

```diff
@@ -38,16 +38,30 @@ def pod_index(name)
     index.chars.map { |c| chars.index(c) }.reduce(0) { |v, c| v * chars.length + c }
   end
 
-  def _get_pods_for_role(role_name)
-    client.get_pods(namespace: namespace, label_selector: "app.kubernetes.io/component=#{role_name}")
+  def _get_sts_image_for_role(role_name)
+    @client_stateful_set.get_stateful_set(role_name, namespace)
+      .spec.template.spec.containers.map(&:image).sort.join("\n")
+  end
+
+  def _get_pods_for_role(role_name, sts_image)
+    client
+      .get_pods(namespace: namespace, label_selector: "app.kubernetes.io/component=#{role_name}")
+      .select do |pod|
+      pod_image = pod.spec.containers.map(&:image).sort.join("\n")
```

Comment:
Missing a level of indent? (`do` shold have an extra level, probably)

Reply:
I would indent it like this:

```
  client.get_pods(namespace: namespace, label_selector: "app.kubernetes.io/component=#{role_name}")
        .select do |pod|
    pod_image = pod.spec.containers.map(&:image).sort.join("\n")
```